app/src/main/java/helium314/keyboard/latin/LatinIME.java

```diff
@@ -1631,13 +1634,21 @@ public void pickSuggestionManually(final SuggestedWordInfo suggestionInfo) {
 
     // This will show either an empty suggestion strip (if prediction is enabled) or
     // punctuation suggestions (if it's disabled).
+    // The toolbar will toggle automatically if the relevant setting is enabled
+    // and there is no text before the cursor or it's the start of a new line.
     @Override
     public void setNeutralSuggestionStrip() {
         final SettingsValues currentSettings = mSettings.getCurrent();
         final SuggestedWords neutralSuggestions = currentSettings.mBigramPredictionEnabled
                 ? SuggestedWords.getEmptyInstance()
                 : currentSettings.mSpacingAndPunctuations.mSuggestPuncList;
         setSuggestedWords(neutralSuggestions);
+        if (hasSuggestionStripView() && currentSettings.mSuggestionsToggleToolbar) {
+            final int codePoint = mInputLogic.mConnection.getCodePointBeforeCursor();
+            if (codePoint == Constants.NOT_A_CODE || codePoint == Constants.CODE_ENTER) {
+                mSuggestionStripView.setToolbarVisibility(true);
+            }
+        }
```

Comment:
You set the visibility right after setting the visibility in `setSuggestedWords`. Such quick successive changes of visibility could be the cause of the flashing of the pinned keys.